pkg/collect/utilisation.go

```diff
@@ -0,0 +1,363 @@
+	if err != nil {
+		return nil, errors.Wrap(err, "Error querying Prometheus for CPU utilization:")
+	}
+	v1api := v1.NewAPI(prometheusClient)
+
+	// Query for CPU utilization
+	cpuQuery := "sum(rate(container_cpu_usage_seconds_total{container!='POD', container!=''}[1h])) by (namespace)"
+	start := time.Now().Add(-1 * time.Duration(c.Collector.Duration) * 24 * time.Hour)
+	end := time.Now()
+	step := time.Duration(c.Collector.SamplingStep) * time.Hour
+	cpuResult, _, err := v1api.QueryRange(context.Background(), cpuQuery, v1.Range{
+		Start: start,
+		End:   end,
+		Step:  step,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "Error querying Prometheus for CPU utilization: ")
+	}
+
+	memoryQuery := "sum(container_memory_working_set_bytes{container!='POD', container!=''}) by (namespace)"
+	memoryResult, _, err := v1api.QueryRange(context.Background(), memoryQuery, v1.Range{
+		Start: start,
+		End:   end,
+		Step:  step,
+	})
+
+	if err != nil {
+		return nil, errors.Wrap(err, "Error querying Prometheus for memory utilization: ")
+	}
+	// Process query result
+	memMatrix, ok := memoryResult.(model.Matrix)
+	if !ok {
+		return nil, errors.New("Invalid query result, expected matrix")
+	}
+
+	// Process query result
+	cpuMatrix, ok := cpuResult.(model.Matrix)
+	if !ok {
+		return nil, errors.New("Invalid query result, expected matrix")
+	}
+	metricsByNamespace := make(map[string]map[string][]model.SamplePair, 0)
+	for _, entry := range memMatrix {
+		namespaceName := string(entry.Metric["namespace"])
+		metricsByNamespace[namespaceName] = make(map[string][]model.SamplePair, 0)
+		metricsByNamespace[namespaceName]["memory"] = entry.Values
+		metricsByNamespace[namespaceName]["cpu"] = []model.SamplePair{}
+	}
+	for _, entry := range cpuMatrix {
+		namespaceName := string(entry.Metric["namespace"])
+		_, ok := metricsByNamespace[namespaceName]
+		if ok {
+			metricsByNamespace[namespaceName]["cpu"] = entry.Values
+		} else {
+			metricsByNamespace[namespaceName] = make(map[string][]model.SamplePair, 0)
+			metricsByNamespace[namespaceName]["cpu"] = entry.Values
+			metricsByNamespace[namespaceName]["memory"] = []model.SamplePair{}
+		}
+	}
+
+	// Construct output.
+	for namespaceName, readingsLists := range metricsByNamespace {
+		jointReadings, err := joinReadings(readingsLists["cpu"], readingsLists["memory"])
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to parse readings: ")
+		}
+
+		payload, err := json.MarshalIndent(jointReadings, "", "  ")
+		if err != nil {
+			return nil, errors.Wrap(err, "Error formatting readings: ")
+		}
+		path := []string{"utilisation", namespaceName}
+		output.SaveResult(c.BundlePath, filepath.Join(path...), bytes.NewBuffer(payload))
+	}
+	stopCh <- struct{}{}
+	return output, nil
+}
+
+func StartPortForwarding(
+	ctx context.Context,
+	restConfig *rest.Config,
+	out output.Output,
```

Comment:
This argument is hardcoded to non interactive output, so I would use nomral gaoling `log` package instead of `dkp-cli-runtime` as the runtime is built for printing CLI messages.